crates/red_knot_project/src/lib.rs

```diff
@@ -71,6 +72,17 @@ pub struct Project {
     #[return_ref]
     pub settings: Settings,
 
+    /// The paths that should be included in this project.
+    ///
+    /// The default (when this list is empty) is to include all files in the project root (that satisfy the configured include and exclude patterns).
+    /// However, it's sometimes desired to only check a subset of the project, e.g. to see
+    /// the diagnostics for a single file or a folder.
+    ///
+    /// This list gets initialized by the paths passed to `knot check <paths>`
+    #[default]
+    #[return_ref]
+    included_paths_list: Vec<SystemPathBuf>,
```

Comment:
nit: we could possibly remove the `_list` suffix to make it `included_paths` ?

Reply:
I've added manual setters and getters that are named `included_paths` that would conflict with the salsa generated once.